### PR TITLE
feat: Add basic e2e tests

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/printeers/trek/issues/7

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  trek init
+
+  trek check
+
+  for file in ../stages/*; do
+    cp "$file" santas_warehouse.dbm
+
+    trek generate "$(basename "$file" | cut -d "-" -f 2 | cut -d "." -f 1)"
+
+    trek check
```
Comment (GeertJohan) [inline tests/run.sh:28]: Should we compare the generated migration to a fixture/"golden" that we store in the repo?

File: tests/run.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -euxo pipefail
+cd "$(dirname "$0")"
+
+rm -rf santas_warehouse
+mkdir -p santas_warehouse
```
[the remark this answers is not in the document]
Comment (provokateurin) [reply tests/run.sh:6]: We want to have the files committed to detect any changes, so it should be inside the project at a fixed path.

File: tests/santas_warehouse/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,5 @@
+FROM ghcr.io/printeers/trek:latest
```
[the remark this answers is not in the document]
Comment (provokateurin) [reply tests/santas_warehouse/Dockerfile:1]: I think we want, in order to be able to detect changes. How are you going to do that otherwise?

**Comment by provokateurin** [comment]
For now we commit everything, but if it turns out to make problems (e.g. the generated image differs) then use the temporary dir approach.

**Pull request merged**